### PR TITLE
Allow closing chat tabs with middle mouse button

### LINKED ISSUE #3418: Close chat tabs with middle mouse button
Similar to the tabs in Google Chrome, I'd like to request adding middle-click to close a chat tab.

### PR DESCRIPTION
**Pull request by hbnrmx**
Closes #3418 
regressed at some point during Chat refactoring.
Originally added in PR #3425

### CODE REVIEW

File: osu.Game/Overlays/Chat/Tabs/ChannelSelectorTabItem.cs [hunk 1]
```diff
@@ -39,5 +40,7 @@ public ChannelSelectorTabChannel()
                 Name = "+";
             }
         }
+
+        protected override bool OnMouseUp(MouseUpEvent e) => false;
```
Comment (Aergwyn) [inline osu.Game/Overlays/Chat/Tabs/ChannelSelectorTabItem.cs]: Why is it overriding this?

Comment (hbnrmx) [reply osu.Game/Overlays/Chat/Tabs/ChannelSelectorTabItem.cs]: Is this unnecessary?
The "+"-Button tab should not be closable.

Comment (Aergwyn) [reply osu.Game/Overlays/Chat/Tabs/ChannelSelectorTabItem.cs]: Ah I see.

Comment (peppy) [reply osu.Game/Overlays/Chat/Tabs/ChannelSelectorTabItem.cs]: It's not required. THere's a check (as you'd expect) to disallow closing in the first place.

**Review by peppy — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**